File: bin/sp-4.4.1/sicstus-4.4.1/library/prologbeans.NET/PBAtom.cs
namespace se.sics.prologbeans
{
	using System;

	/// <summary> <code>PBAtom</code> is the representation of Prolog atoms.</summary>
	
	class PBAtom:PBAtomic
	{
		override sealed internal int Type {
			get {
				return ATOM;
			}
		}

		override sealed public bool Atom {
			get {
				return true;
			}
		}

		/// <summary> Creates a new <code>PBAtom</code> instance with the specified name.</summary>
		internal PBAtom (string name) : base (name)
		{
		}

		internal override string toPrologString ()
		{
			return stuffAtom (Name);
		}

		public override string ToString ()
		{
			return Name;
		}
	}
}

File: bin/sp-4.4.1/sicstus-4.4.1/se/sics/prologbeans/PBAtom.java
/**
 * Copyright (c) 2003 SICS AB. All rights reserved.
 */
package se.sics.prologbeans;

/**
 * <code>PBAtom</code> is the representation of Prolog atoms.
 */

class PBAtom extends PBAtomic
{
    /**
     * Creates a new <code>PBAtom</code> instance with the specified name.
     */
    PBAtom(String name)
    {
        super(name);
    }

    int getType()
    {
        return ATOM;
    }

    public boolean isAtom()
    {
        return true;
    }

    String toPrologString()
    {
        return stuffAtom(name);
    }

    public String toString()
    {
        return name;
    }

}
